Fix regression test for visual prompting & Add `ResizeToLongestEdge`

sungchul1 opened the pull request:
### Summary

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

This PR includes:
- [x] Fix regression test for visual prompting task (reported by @chuneuny-emily)
    - visual prompting task needs to remove unused parameters which is already assigned in the base config
- [x] Add `ResizeToLongestEdge` for square images
    - tvt_v2 `Resize` doesn't support resizing images to the longest edge for square images
    - [x] unit test
- [x] Refactoring

### How to test

<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated. -->

### Checklist

<!-- Put an 'x' in all the boxes that apply -->

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

sungchul1 committed:
Fix regression test

sungchul1 committed:
Add `ResizeToLongestEdge` for square image & refactoring

sovrasov commented on src/otx/core/data/transform_libs/torchvision.py:
```diff
@@ -42,22 +42,59 @@ def _perturb_bounding_boxes(self, inpt: torch.Tensor, offset: int) -> torch.Tens
+        return self._call_kernel(F.pad, inpt, padding=params["padding"], fill=0, padding_mode="constant")
+
+
+class ResizetoLongestEdge(tvt_v2.Transform):
```
FYI any custom resize type might complicate integration with ModelAPI (if such is not planned, then this comment is not relevant)

sungchul1 replied:
It works like resize part of `fit_to_window` in ModelAPI.

vinnamkim reviewed (dismissed):
LGTM.

sungchul1 committed:
Merge branch 'v2' into fix_v2_regression_test

sungchul1 committed:
Disable deterministic
RuntimeError: cumsum_cuda_kernel does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True)'. You can turn off determinism just for this operation, or you can use the 'warn_only=True' option, if that's acceptable for your application. You can also
file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation.

sungchul1 committed:
Update config

sungchul1 committed:
Update num_workers

sungchul1 committed:
Merge branch 'v2' into fix_v2_regression_test
